### DISCUSSION TITLE
headspace.com getting "Invalid audio stream."

### DISCUSSION THREAD

**Discussion by aacecandev** [aclap-dev/video-downloadhelper#303] [Help and Q&A]
### Your question

Hi,

I'm trying to download from this site, form where I've been able to download before both audio and video. My intention is to get the output as MP3.

The extensions recognises that there's an audio stream, but when trying to download it, I'm only getting 3 options, they're look the same "MP3 - HLS Streaming"

When I click to download it, the download starts but in a few seconds it stops and shows this error

```
SideDownload error:
[mp3 @ 0x5a32140] Invalid audio stream. Exactly one MP3 audio stream is required.
[out#0/mp3 @ 0x5a32740] Could not write header (incorrect codec parameters ?): Invalid argument
Error opening output file /home/alejandro.aceituna/dwhelper/Audio Player-01.mp3.
Error opening output files: Invalid argument
```

I've tried to adjust the setup but I've not found any option that I could be potentially missing. 

Could you please give me a hand?

### Addon & CoApp version and details

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: true, browser: chrome
lang: en-US
license: unneeded
platform: x86-64 linux
UA: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "/home/alejandro.aceituna/.local/share/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/alejandro.aceituna",
  "converterBinary": "/home/alejandro.aceituna/.local/share/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360"
}

**Comment by GrampaWildWilly** [comment]
One option you're missing is to tell us at least one URL of a web page on which one of your problematic videos is hosted.

**Reply by aacecandev** [reply]
hi @GrampaWildWilly,

I specified it in this issue head title :) But for clarification, they're hosted in `headspace.com`

**Reply by Mattmjs** [reply]
Is it a video that you want audio from ? share the hit details information for something your after.

**Reply by aacecandev** [reply]
hi @Mattmjs

it is a streaming audio from which I want to get the audio, but I am not sure if that's what you asked me

**Reply by Mattmjs** [reply]
I was going to ask that but left it out, there are 3 mp3 files detected on the homepage. Do you have success on these files ?
Edit: those should work but I see in your original post you want something which is "MP3 - HLS Streaming".
If you can provide the hit details information I can test if it works or not.

**Reply by aacecandev** [reply]
hi @Mattmjs,

nevermind, I can't say how but it has started working. I was being offered with 3 identical options "MP3 - HLS Streaming" but after clicking any of them, the error in the OP was shown.

I'm closing this issue as resolved
